lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,29 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "parameter MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
```

Comment:
The problem if you do that is that when `A = {}, B = { a: {} }`, when you do `A.merge!(B)` and subsequently modify `B`, `A` will be modified as well. I can't really think of a use case where one would do modifications, but it might be confusing.